.github/workflows/check-odo.yml

```diff
@@ -0,0 +1,43 @@
+name: check-odo
+
+on: [workflow_dispatch]
+
+jobs:
+  check-odo-repo:
+    runs-on: ubuntu-latest
+    env:
+        TOOL_REPO: redhat-developer/odo
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check Out Code
+        uses: actions/checkout@v2
+      - name: Get latest ODO version
+        run: |
+          echo "REPO_ODO_VERSION=$(cat src/tools.json | jq -r .odo.version)" >> $GITHUB_ENV
+          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName,url)
+          echo "LATEST_TOOL_RELEASE=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName | sed 's|v||')" >> $GITHUB_ENV
+          echo "LATEST_TOOL_URL=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .url)" >> $GITHUB_ENV
+      - name: Find existing PR for ODO version
+        run: |
+          echo PR_URL=$(gh pr --repo ${{ github.repository }} list --state all --search "update odo ${{env.LATEST_TOOL_RELEASE}} in:title" --json url) >> $GITHUB_ENV
+      - name: Update src/tools.json with latest odo version
+        if: ${{ (env.LATEST_TOOL_RELEASE != env.REPO_ODO_VERSION) && (env.PR_URL == '') }}
```

Comment:
As discussed `env.PR_URL` returns array, So we can handle through length check

Reply:
I just called it PR_EXISTS, just to make it clear we're only using the value for a boolean check.